lib/dynamics_crm/client.rb

```diff
@@ -141,7 +141,13 @@ def rollup(target_entity, query, rollup_type="Related")
         })
     end
 
-    def retrieve_multiple(entity_name, criteria=[], columns=[], operator=nil)
+    def retrieve_multiple_query(query_expression)
```

Comment:
I may have `retrieve_multiple` support both.  I would check the first parameter (`entity_name`) to see if it is an `instance_of?(QueryExpression)` and use that, then we wouldn't need the separate method.